aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/util/CognitoAuthExceptionConverter.java

```diff
@@ -0,0 +1,109 @@
+import com.amazonaws.services.cognitoidentityprovider.model.TooManyFailedAttemptsException;
+import com.amazonaws.services.cognitoidentityprovider.model.UnexpectedLambdaException;
+import com.amazonaws.services.cognitoidentityprovider.model.UserNotConfirmedException;
+import com.amazonaws.services.cognitoidentityprovider.model.UserNotFoundException;
+import com.amazonaws.services.cognitoidentityprovider.model.UsernameExistsException;
+
+/**
+ * Convert AWS Cognito Exceptions to AuthExceptions.
+ */
+public final class CognitoAuthExceptionConverter {
+
+    /**
+     * Dis-allows instantiation of this class.
+     */
+    private CognitoAuthExceptionConverter() {}
+
+    /**
+     * Lookup method to convert AWS Cognito Exception to AuthException.
+     * @param error Exception thrown by AWSMobileClient
+     * @return AuthException
+     */
+    public static AuthException lookup(Exception error) {
+        if (error instanceof UserNotFoundException) {
+            return new AuthException.UserNotFoundException(error);
+        }
+
+        if (error instanceof UserNotConfirmedException) {
+            return new AuthException.UserNotConfirmedException(error);
+        }
+
+        if (error instanceof UsernameExistsException) {
+            return new AuthException.UsernameExistsException(error);
+        }
+
+        if (error instanceof AliasExistsException) {
+            return new AuthException.AliasExistsException(error);
+        }
+
+        if (error instanceof InvalidPasswordException) {
+            return new AuthException.InvalidPasswordException(error);
+        }
+
+        if (error instanceof InvalidParameterException) {
+            return new AuthException.InvalidParameterException(error);
+        }
+
+        if (error instanceof ExpiredCodeException) {
+            return new AuthException.CodeExpiredException(error);
+        }
+
+        if (error instanceof CodeMismatchException) {
+            return new AuthException.CodeMismatchException(error);
+        }
+
+        if (error instanceof CodeDeliveryFailureException) {
+            return new AuthException.CodeDeliveryFailureException(error);
+        }
+
+        if (error instanceof LimitExceededException) {
+            return new AuthException.LimitExceededException(error);
+        }
+
+        if (error instanceof ResourceNotFoundException) {
+            return new AuthException.ResourceNotFoundException(error);
+        }
+
+        if (error instanceof TooManyFailedAttemptsException) {
+            return new AuthException.FailedAttemptsLimitExceededException(error);
+        }
+
+        if (error instanceof UnexpectedLambdaException) {
+            return new AuthException.LambdaException(error);
+        }
+
+        if (error instanceof PasswordResetRequiredException) {
+            return new AuthException.PasswordResetRequiredException(error);
+        }
+
+        return null;
```

Comment:
You could further simplify your code in the plugin, by providing a default exception, here:
```suggestion
        return new AuthException(fallbackMessage, error, "See attached exception for more details.");
```
Then, you'd only need to pass in an additional `fallbackMessage` when you call the lookup:
```java
AuthException mapped =
    CognitoAuthExceptionConverter.lookup("Fallback message for some situation, here.", error);
```

Note: this is just a suggestion. You can choose not to do this.

Reply:
This is a very good suggestion!